### PR TITLE
[BIOMAGE-1044] Add a needsRunning flag to gem2s pipeline status

### PR DESCRIPTION
**Pull request by xverges**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1044

#### Link to staging deployment URL 
https://ui-martinfosco-ui319-api147-p.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/api/pull/147
https://github.com/biomage-ltd/pipeline/pull/96

#### Anything else the reviewers should know about the changes here
Backend status now returns a `needsRunning` flag for gem2s. `needsRunning` is true if 
* pipeline never executed
* pipeline execution ended in error
* pipeline execution was successful but a was launched with different params 
# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [x] Photo of a cute animal attached to this PR
brown-mesh-sea-star

### CODE REVIEW

File: src/components/data-management/AnalysisModal.jsx [hunk 1]
```diff
@@ -29,12 +29,16 @@ const NewExperimentModal = (props) => {
 
   const [experimentsList, setExperimentsList] = useState([]);
   const [numFieldsEditing, setNumFieldsEditing] = useState(0);
+  const [isWorking, setIsWorking] = useState(false);
```
Comment (aerlaut) [inline src/components/data-management/AnalysisModal.jsx:32]: I don't understand what this `isWorking` state is for. It's `true` when the modal is not visible, and it disables the launch experiment button, which is not visible as the modal is not visible.

**Review by aerlaut — COMMENTED** [review]
Asked comment for clarification

File: src/components/data-management/AnalysisModal.jsx (hunk 1 above)
Comment (cosa65) [reply src/components/data-management/AnalysisModal.jsx:32]: Yes, I am not sure either, it seems very distantly related to the rest of ticket

Comment (cosa65) [reply src/components/data-management/AnalysisModal.jsx:32]: My guess is it is some way to display some kind of loading in the ui between launching an analysis and the ui switching to data processing

Comment (cosa65) [reply src/components/data-management/AnalysisModal.jsx:32]: Yes, I would call it something else probably and perhaps check if it really needs to depend on visibility (I think that part is unnecessary)

Comment (xverges) [reply src/components/data-management/AnalysisModal.jsx:32]: 👋 The goal of `isWorking` is to disable the launch buttons after it has been clicked; under some circumstances, the user has no visual indication that his click had effect. When the dialog becomes visible, we make sure that `isWorking` is reset.

Comment (cosa65) [reply src/components/data-management/AnalysisModal.jsx:32]: Thanks!!

**Review by aerlaut — APPROVED** [review]
LGTM

**Pull request merged**